### PR TITLE
Add support for Alpine on LGBM

### PR DESCRIPTION
**Pull request by Renato2000**
This commit makes it possible to use LGBM on nodes with AMD64 architecture using Alpine

### CODE REVIEW

File: make.sh [hunk 1]
```diff
@@ -76,9 +76,11 @@ if [[ "$ARCH_BUILD" != "single" ]]; then
     bash docker/make_docker_image.sh arm64
 fi
 
+bash docker/make_docker_image.sh alpine
+
 echo_stage "Launching AMD64 container..."
 container_amd64=$(docker run -e LIGHTGBM_REPO_URL -t -d lightgbm-ci-build-env-amd64)
-docker cp docker/make_lightgbm.sh $container_amd64:/lightgbm
+docker cp docker/make_lightgbm.sh $container_amd64:/lightgbm/make_lightgbm.sh
```
Comment (artpdr) [inline make.sh]: Why is this change necessary?

File: make.sh [hunk 2]
```diff
@@ -110,6 +121,12 @@ if [[ "$ARCH_BUILD" != "single" ]]; then
     docker cp $container_arm64:/usr/lib/aarch64-linux-gnu/libgomp.so.1.0.0 build/arm64
 fi
 
+FROM_ALPINE="$container_alpine:/lightgbm/LightGBM"
+docker cp $FROM_ALPINE/lib_lightgbm.so build/alpine
```
Comment (artpdr) [inline make.sh]: The README needs to be updated to account for this new build/alpine. And probably this new build/alpine should be inside the amd64 folder. Please check with other members the best structure having in mind that changing the directory structure may be a breaking change and will require a major release.

**Review by artpdr — CHANGES_REQUESTED** [review]

File: make.sh (hunk 1 above)
Comment (Renato2000) [reply make.sh]: That line was giving errors on the Alpine version so I also changed it for amd64. However it was actually not necessary and so I reverted it in the last commit.

File: make.sh (hunk 2 above)
Comment (Renato2000) [reply make.sh]: I moved the alpine build into the amd64 folder. So, the new structure is: build/amd64/glibc, build/amd64/alpine and build/arm64. I will also update the README in the next commit.

**Review by artpdr — APPROVED** [review]

File: docker/lightgbm-ci-build-env-alpine/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM alpine:3.18.4
+
+RUN apk add clang git cmake make gcc g++ tar zip curl openjdk8 bash
```
Comment (gandola) [inline docker/lightgbm-ci-build-env-alpine/Dockerfile]: is openmp being installed and used?

File: README.md [hunk 4]
```diff
@@ -43,9 +43,14 @@ This is the output:
...
 build
 ├── amd64
-│   ├── libgomp.so.1.0.0
-│   ├── lib_lightgbm.so
-│   └── lib_lightgbm_swig.so
+│   ├── glibc
+│   │   ├── libgomp.so.1.0.0
+│   │   ├── lib_lightgbm.so
+│   │   └── lib_lightgbm_swig.so
+│   ├── alpine
```
Comment (gandola) [inline README.md]: don't make it `alpine`, `musl` is the correct term here.

File: docker/make_docker_image.sh [hunk 5]
```diff
@@ -29,6 +29,10 @@ case "$1" in
         echo "Building docker image for AMD64"
         docker build -t lightgbm-ci-build-env-amd64 lightgbm-ci-build-env-amd64
         ;;
+    alpine)
+        echo "Building docker image for Alpine"
```
Comment (gandola) [inline docker/make_docker_image.sh]: Alpine with Musl

File: docker/lightgbm-ci-build-env-alpine/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,18 @@
+FROM alpine:3.18.4
```
Comment (gandola) [inline docker/lightgbm-ci-build-env-alpine/Dockerfile:1]: What are the software versions on this alpine version? clang, cmake, etc? 

Lets use Zulu JDK which is the one being used by other builds.

File: README.md (hunk 4 above)
Comment (Renato2000) [reply README.md]: I changed it to musl in the last commit.

File: docker/lightgbm-ci-build-env-alpine/Dockerfile (hunk 6 above)
Comment (Renato2000) [reply docker/lightgbm-ci-build-env-alpine/Dockerfile:1]: The software versions are:
  
  cmake=3.26.5
  make=4.4.1
  gcc  and g++: 12.2.1
  
  In the last commit I specified the version when running apk add. Also, clang was removed because it was not necessary.

Comment (Renato2000) [reply docker/lightgbm-ci-build-env-alpine/Dockerfile:1]: I also installed the same version of Zulu JDK that is being used on the other builds.

File: docker/lightgbm-ci-build-env-alpine/Dockerfile (hunk 3 above)
Comment (Renato2000) [reply docker/lightgbm-ci-build-env-alpine/Dockerfile]: I think openmp in being installed alongside g++. Since it also installs gcc v12.2.1 and according to this [gcc.gnu.org/wiki/openmp](https://gcc.gnu.org/wiki/openmp), it fully supports OpenMP 4.5.

**Comment by gandola** [comment]
LGTM!
